internal/nexodus/nexodus.go

```diff
@@ -435,6 +441,41 @@ func (ax *Nexodus) Start(ctx context.Context, wg *sync.WaitGroup) error {
 	ax.logger.Infof("Successfully registered device with UUID: [ %+v ] into organization: [ %s (%s) ]",
 		modelsDevice.Id, ax.org.Name, ax.org.Id)
 
+	if modelsDevice.SecurityGroupId != uuid.Nil.String() {
+		err = util.RetryOperation(ctx, retryInterval, maxRetries, func() error {
+			ax.securityGroup, resp, err = ax.client.SecurityGroupApi.GetSecurityGroup(ctx, organizations[0].Id, organizations[0].SecurityGroupId).Execute()
+			if err != nil {
+				if strings.Contains(err.Error(), securityGroupNotFound.Error()) {
+					ax.logger.Debugf("security group returned a 404")
+					return nil
+				} else {
+					if resp != nil {
+						ax.logger.Warnf("get security group error - retrying error: %v header: %+v", err, resp.Header)
+						return err
+					}
+					if err != nil {
+						ax.logger.Warnf("get security group error - retrying error: %v", err)
+						return err
+					}
+				}
+			}
+			ax.logger.Debugf("organization security group: %s", ax.securityGroup.Id)
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("join error %w", err)
+		}
+		// apply the initial security group rules
+		if runtime.GOOS == Linux.String() {
+			ax.logger.Debugf("Security group: %+v", ax.securityGroup)
+			if err := ax.processSecurityGroupRules(); err != nil {
+				return err
+			}
+		}
+	} else {
+		ax.logger.Debug("organization security group is empty")
+	}
```

Comment:
Why is this needed over just relying on the reconcile goroutine to set up security groups?

Reply:
I think I have it trimmed down to where reconcile can be used for a first time shot. 

The default tables still need to be setup since there isn't going to be a diff to trigger any rule changes in reconcile.  ModelsSecurityGroup still needs to be initialized before going into reconcile or else it is nil. 

Im pretty comfortable with the following:

```go
	// apply the default security group rules
	ax.securityGroup = &public.ModelsSecurityGroup{}
	ax.securityGroup.Id = modelsDevice.SecurityGroupId
	if runtime.GOOS == Linux.String() {
		if err := ax.processSecurityGroupRules(); err != nil {
			return err
		}
	}

	util.GoWithWaitGroup(wg, func() {
		// kick it off with an immediate reconcile
		ax.reconcileDevices(ctx, options)
		ax.reconcileSecurityGroups(ctx)
```

Reply:
OK, I'll follow up on this after merging. https://github.com/nexodus-io/nexodus/issues/1132